### PR TITLE
Generate test chains with valid chain value pools

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Zebra needs to generate test chains with valid chain value pools, so it can pass existing tests, and test chain value pool validation.

## Solution

- Add a `fix_chain_value_pools` method in `transaction::arbitrary`
- Use this method to fix and update the chain value pools for generated chains in `block::arbitrary`
- Reduce scaling workaround, so it's based on the actual number of blocks and transactions in the tests
- Split `fix_remaining_value` into smaller methods
- Remove unused methods

This is part of ticket #1895, but doesn't close that ticket.

## Review

@oxarbitrage can review this PR, it blocks testing for PR #2599.

### Reviewer Checklist

  - [x] Code makes sense
  - [x] Documentation makes sense
  - [x] Existing tests pass

---
This change is [Reviewable](https://reviewable.io/reviews/zcashfoundation/zebra/2597)

### CODE REVIEW

File: zebra-chain/src/block/arbitrary.rs [hunk 1]
```diff
@@ -43,6 +47,17 @@ use super::*;
...
+/// The chain length for most zebra-state proptests.
+///
+/// Most generated chains will contain transparent spends at or before this height.
+///
+/// This height was chosen as a tradeoff between chains with no transparent spends,
+/// and chains which spend outputs created by previous spends.
+///
+/// See [`block::arbitrary::PREVOUTS_CHAIN_HEIGHT`] for details.
+pub const MAX_PARTIAL_CHAIN_BLOCKS: usize =
+    MIN_TRANSPARENT_COINBASE_MATURITY as usize + PREVOUTS_CHAIN_HEIGHT;
```
Comment (teor2345) [inline zebra-chain/src/block/arbitrary.rs:59]: This code is moved from zebra-state

File: zebra-chain/src/transparent.rs [hunk 2]
```diff
@@ -35,6 +35,15 @@ use crate::{
...
 use std::{collections::HashMap, iter};
 
+/// The maturity threshold for transparent coinbase outputs.
+///
+/// "A transaction MUST NOT spend a transparent output of a coinbase transaction
+/// from a block less than 100 blocks prior to the spend. Note that transparent
+/// outputs of coinbase transactions include Founders' Reward outputs and
+/// transparent Funding Stream outputs."
+/// [7.1](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus)
+pub const MIN_TRANSPARENT_COINBASE_MATURITY: u32 = 100;
```
Comment (teor2345) [inline zebra-chain/src/transparent.rs:45]: This code is moved from zebra-state

File: zebra-chain/src/transaction/arbitrary.rs [hunk 3]
```diff
@@ -218,59 +220,168 @@ impl Transaction {
         }
     }
 
-    /// Fixup non-coinbase transparent values and shielded value balances,
-    /// so that this transaction passes the "remaining transaction value pool" check.
+    /// Fixup transparent values and shielded value balances,
+    /// so that transaction and chain value pools won't overflow MAX_MONEY.
     ///
-    /// Returns the remaining transaction value.
+    /// These fixes are applied to coinbase and non-coinbase transactions.
```
Comment (teor2345) [inline zebra-chain/src/transaction/arbitrary.rs:226]: This file might be easier to review in Split diff mode.
(The Split diffs setting is under the ⚙️ icon at the top of the file view.)

https://github.blog/2014-09-03-introducing-split-diffs/

**Review by teor2345 — COMMENTED** [review]
Just a few explanations for the changes in this PR

**Review by oxarbitrage — APPROVED** [review]

**Pull request merged**